app/vmalert/notifier/template.go

```diff
@@ -31,6 +33,67 @@ import (
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promutils"
 )
 
+// go template execution fails when it's tree is empty
+const defaultTemplate = `{{- define "default.template" -}}{{- end -}}`
+
+type textTemplate struct {
+	current     *textTpl.Template
+	replacement *textTpl.Template
+}
+
+var masterTmpl textTemplate
+
+// loads templates from multiple globs specified in pathPatterns and either
+// sets them directly to current template if it's undefined or with overwrite=true
+// or sets replacement templates and adds templates with new names to a current
+func LoadTemplates(pathPatterns []string, overwrite bool) error {
+	var err error
+	tmpl := textTpl.New("").Option("missingkey=zero").Funcs(templateFuncs())
+	tmpl = textTpl.Must(tmpl.Parse(defaultTemplate))
```

Comment:
`.Must` can panic during the hot reload and will crash the process. The error needs to be handled here.

Reply:
This template is constant, it should not fail and execution should not continue if it has failed.